hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -202,6 +202,30 @@ static inline void vlapic_build_x2apic_id(struct acrn_vlapic *vlapic)
 	lapic->ldr.v = (cluster_id << 16U) | (1U << logical_id);
 }
 
+static inline uint32_t vlapic_find_isrv(const struct acrn_vlapic *vlapic)
+{
+	const struct lapic_regs *lapic = &(vlapic->apic_page);
+	uint32_t i, val, bitpos, isrv = 0U;
+	const struct lapic_reg *isrptr;
+
+	if (vlapic->isr_nr != 0U) {
+		isrptr = &lapic->isr[0];
+
+		/* i ranges effectively from 7 to 1 */
+		for (i = 8U; i > 1U; ) {
+			i--;
```

Comment:
why not check isr[0]? I thought only 0x0 to 0xf are reserved for exception.

Reply:
SDM, Vol 3, Chap 10.5.2 Valid Interrupt Vectors
“The Intel 64 and IA-32 architectures define 256 vector numbers, ranging from 0 through 255 (see Section 6.2, “Exception and Interrupt Vectors”). Local and I/O APICs support 240 of these vectors (in the range of 16 to 255) as valid interrupts.  ......  The Intel 64 and IA-32 architectures reserve vectors 16 through 31 for predefined interrupts, exceptions, and Intel-reserved encodings (see Table 6-1). However, the local APIC does not treat vectors in this range as illegal.”
IMHO, in spite of vector from 16  through 31 is valid, for these reserve vectors, we should not deliver them to LAPIC.

Reply:
for （ i = 7U; i > 0U; i--)  is better than this one.

Reply:
Yes, will change like this.